### ISSUE TITLE
Fix the root cause why `byte` has to be used instead of `bool` for one C# interop call to C++

### ISSUE THREAD

**Issue by hhyyrylainen** [Revolutionary-Games/Thrive#5676]
for some reason a single method call for checking if body is detached gets the wrong result to C# when the extern definition has `bool` instead of `byte` in it. This is super bizarre. And very worrying if other function calls also have the same problem or not...

**Comment by xfractalino** [comment]
This seems to be related by the fact that bool can be implementation-specific. I think here bool can actually be marshaled as a 32-bit integer instead of a single byte as expected. `byte` forces using 8 bits only.

**Comment by hhyyrylainen** [comment]
Oh yeah, I remember now that one of the newer interop calls added looks like this:

```c#
    [SuppressGCTransition]
    [DllImport("thrive_native")]
    [return: MarshalAs(UnmanagedType.I1)]
    internal static extern bool ArmWaitForEvent();
```

So that's likely the reason and all `bool` return values will need the same attribute fix (`MarshalAs`).

**Comment by xfractalino** [comment]
Exactly, that's what I was going to do on the `PhysicsBodyIsDetached` that is mentioned in the issue.
